### PR TITLE
Homework 8

### PR DESCRIPTION
**Pull request by Jan84**
Check out homework 8

### CODE REVIEW

File: html/productsPage.html [hunk 1]
```diff
@@ -11,8 +11,13 @@
...
+				<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg id="svg-cart" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="24" height="24" viewBox="0 0 24 24"><path id="svg-path" d="M17,18A2,2 0 0,1 19,20A2,2 0 0,1 17,22C15.89,22 15,21.1 15,20C15,18.89 15.89,18 17,18M1,2H4.27L5.21,4H20A1,1 0 0,1 21,5C21,5.17 20.95,5.34 20.88,5.5L17.3,11.97C16.96,12.58 16.3,13 15.55,13H8.1L7.2,14.63L7.17,14.75A0.25,0.25 0 0,0 7.42,15H19V17H7C5.89,17 5,16.1 5,15C5,14.65 5.09,14.32 5.24,14.04L6.6,11.59L3,4H1V2M7,18A2,2 0 0,1 9,20A2,2 0 0,1 7,22C5.89,22 5,21.1 5,20C5,18.89 5.89,18 7,18M16,11L18.78,6H6.14L8.5,11H16Z" /></svg>
 				<div id="circle-counter">0</div>
+				<div id="bag-popup" class="hidden">
+					<div class="top-arrow"></div>
+					<div id="bag-total-text">Total: </div>
+					<div id="total-price">0 Lei</div>
```
Comment (ionutmiftode) [inline html/productsPage.html]: create another `<div>Lei</div>`

File: js/productsPage.js [hunk 2]
```diff
@@ -22,14 +22,31 @@ uniqueBrands.forEach(function(product) {
 var bagCounter = 0;
 var circle = document.getElementById("circle-counter");
 
+// Add to bag (number of products and total price)
 var addToBagClickHandler = function(e) {
 	if(e.target.type == 'submit') {
```
Comment (ionutmiftode) [inline js/productsPage.js]: maybe we will add other buttons to our products. Let's check if it has `add-to-bag` class

File: js/productsPage.js [hunk 3]
```diff
@@ -22,14 +22,31 @@ uniqueBrands.forEach(function(product) {
...
+		bag.updateTotal();
 		bagCounter++;
 		circle.textContent = bagCounter;
 	}
 }
 sectionProducts.addEventListener('click', addToBagClickHandler);
 
+// Show/hide cart popup
+document.addEventListener('click', function(e) {
+	if(e.target.id == 'svg-cart' || e.target.id == 'svg-path') {
```
Comment (ionutmiftode) [inline js/productsPage.js]: we want to close the popup even when we click the bag icon

File: js/productsPage.js [hunk 4]
```diff
@@ -22,14 +22,31 @@ uniqueBrands.forEach(function(product) {
...
+		bag.products.push(product);
+		bag.updateTotal();
 		bagCounter++;
 		circle.textContent = bagCounter;
 	}
 }
 sectionProducts.addEventListener('click', addToBagClickHandler);
 
+// Show/hide cart popup
+document.addEventListener('click', function(e) {
```
Comment (ionutmiftode) [inline js/productsPage.js]: Let's add the event handler `function(e) {}` as `onClick` method on our `bag`

File: js/script.js [hunk 5]
```diff
@@ -5,14 +5,31 @@ var bagCounter = 0;
...
+		bag.products.push(product);
+		bag.updateTotal();
 		bagCounter++;
 		circle.textContent = bagCounter;
 	}
 }
 sectionProducts.addEventListener('click', addToBagClickHandler);
 
+// Show/hide cart popup
+document.addEventListener('click', function(e) {
```
Comment (ionutmiftode) [inline js/script.js]: create a `common.js` file and add the duplicated code in there

File: js/bag.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+
+var bag = {
```
Comment (ionutmiftode) [inline js/bag.js]: create a new method `add` and move the logic from `addToBagClickHandler`

Comment (ionutmiftode) [inline js/bag.js]: create a `total: 0` property that will retain the total price

File: js/bag.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+	popup: document.getElementById("bag-popup"),
+	products: [],
+	getTotalPrice: function() {
```
Comment (ionutmiftode) [inline js/bag.js]: remove `getTotalPrice`

File: js/bag.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+		var price = 0;
+		for (var i = 0; i < this.products.length; i++) {
+			price += this.products[i].price;
+		}
+		return price;
+	},
+	updateTotal: function() {
+		var totalPrice = this.getTotalPrice();
```
Comment (ionutmiftode) [inline js/bag.js]: add the `price` of the last element added to `products` into `total`

File: html/productsPage.html (hunk 1 above)
Comment (Jan84) [reply html/productsPage.html]: Ok, dar care este motivul pentru care trebuie un div separat?

Comment (ionutmiftode) [reply html/productsPage.html]: Lei este static, nu ar trebui sa-l updatam tot timpul cand se schimba pretul

File: js/bag.js (hunk 6 above)
Comment (Jan84) [reply js/bag.js]: Can you give me more details? total: 0 and where I will retain total price? I need a method I think, not a property..

Comment (ionutmiftode) [reply js/bag.js]: with 0 it will be initialized and every time when an item is added total will be updated

Comment (Jan84) [reply js/bag.js]: I got it, problem solved :)
Thank you!

File: js/productsPage.js (hunk 4 above)
Comment (Jan84) [reply js/productsPage.js]: Here I have some problems, I`m not sure if I understood correct:
1. I made function clickOnBag(e) { same code } 
2. I used in bag html: onclick="javascript:clickOnBag();" but it seems that e is not defined
I need to change the code and not use 'e' ? I need some details to understand what I have to do.

**Pull request merged**